lib/fennec/turn.ex

```diff
@@ -2,13 +2,19 @@ defmodule Fennec.TURN do
   @moduledoc false
   # This module defines a struct used as TURN protocol state.
 
-  defstruct allocation: nil, permissions: %{}, channels: [],
-            nonce: nil, realm: nil
+  alias Jerboa.Format
+  alias Fennec.TURN.Channel
+
+  defstruct allocation: nil, permissions: %{}, channels: {%{}, %{}},
+    nonce: nil, realm: nil
+
+  @permission_lifetime 5 * 60 # MUST be 5mins
 
   @type t :: %__MODULE__{
     allocation: nil | Fennec.TURN.Allocation.t,
     permissions: %{peer_addr :: Fennec.ip => expiration_time :: integer},
-    channels: [],
+    channels: {%{peer :: Fennec.address => Channel.t},
+               %{Format.channel_number  => Channel.t}},
```

Comment:
I can see why we keep these together. Another option would be a table (one data structure):

```Elixir
[{Fennec.address, Format.channel_number, Channel.t}, ...]
```

Then we can fetch through address or number with `List.keyfind/4` or similar. It'd be easier to keep it all consistent this way.